### PR TITLE
Arrow buffer Size Option

### PR DESCRIPTION
**Pull request by pdet**
This PR allows users to set the Arrow Buffer Size used for Strings, BLOBS, UUIDS.

It can be set by changing the `arrow_large_buffer_size` option.
```sql
SET arrow_large_buffer_size=True
``` 

By default, it also changes DuckDB to output regular buffer sizes.

Obs: Currently, the extension building CI is failing. The  Arrow-IPC extension PR https://github.com/duckdblabs/arrow/pull/12 should be merged first, and the extension retagged to fix all the extension CI.

OBS2: I consider this somewhat of a bug fix, but I can retag it to feature if desired.

### CODE REVIEW

**Comment by Tishj** [comment]
To preemptively avoid issues like <https://github.com/duckdb/duckdb/pull/7258> can we not use `timezone` as a variable name and instead use `time_zone` ?

File: src/include/duckdb/common/arrow/arrow_options.hpp [hunk 1]
```diff
@@ -13,6 +13,13 @@ namespace duckdb {
...
 struct ArrowOptions {
-	ArrowOffsetSize offset_size = ArrowOffsetSize::LARGE;
+	explicit ArrowOptions(ArrowOffsetSize offset_size_p) : offset_size(offset_size_p) {
+	}
+	ArrowOptions(ArrowOffsetSize offset_size_p, string timezone_p) : offset_size(offset_size_p), timezone(timezone_p) {
+	}
+	ArrowOptions() {
+	}
+	ArrowOffsetSize offset_size = ArrowOffsetSize::REGULAR;
+	string timezone = "UTC";
```
Comment (Tishj) [inline src/include/duckdb/common/arrow/arrow_options.hpp:23]: To preemptively avoid issues like <https://github.com/duckdb/duckdb/pull/7258> can we not use `timezone` as a variable name and instead use `time_zone` ?

**Review by Tishj — COMMENTED** [review]
Looks good to me, could maybe use a couple extra tests since this changes a lot of different places where arrow is produced.

Of the top of my head, a scalar UDF test that makes sure the strings passed to the UDF are of the correct type could be nice

**Comment by carlopi** [comment]
Hi @pdet!
I guess WebAssembly CI will fail, given that the is an incompatibility (chicken & egg, ...), I can do the changes WebAssembly side, I'd say this looks stable, I will get the changes in duckdb-wasm ready, and then we merge the two PRs them more or less together (so subsequent CI will not fail).

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**